src/support/global.c

```diff
@@ -80,6 +80,94 @@ __endian_check(void)
     return (EINVAL);
 }
 
+/*
+ * __reset_thread_tick --
+ *     Reset the OS task time slice to raise the probability of uninterrupted run afterwards.
+ */
+static void
+__reset_thread_tick(void)
+{
+    /*
+     * We could use __wt_yield() here but simple yielding doesn't seem to always reset the task's
+     * time slice. Sleeping for a short time does a better job.
+     */
+    __wt_sleep(0, 10);
+}
+
+/*
+ * __compare_uint64 --
+ *     uint64_t comparison function.
+ */
+static int
+__compare_uint64(const void *a, const void *b)
+{
+    return (int)(*(uint64_t *)a - *(uint64_t *)b);
+}
+
+/*
+ * __get_epoch_call_ticks --
+ *     Returns how many ticks it takes to call __wt_epoch at best and on average.
+ */
+static void
+__get_epoch_call_ticks(uint64_t *epoch_ticks_min, uint64_t *epoch_ticks_avg)
+{
+#define EPOCH_CALL_CALIBRATE_SAMPLES 50
+    uint64_t duration[EPOCH_CALL_CALIBRATE_SAMPLES];
+
+    __reset_thread_tick();
+    for (int i = 0; i < EPOCH_CALL_CALIBRATE_SAMPLES; ++i) {
+        struct timespec clock1;
+        uint64_t tsc1 = __wt_rdtsc();
+        __wt_epoch(NULL, &clock1);
+        uint64_t tsc2 = __wt_rdtsc();
+        duration[i] = tsc2 - tsc1;
+    }
+    __wt_qsort(duration, EPOCH_CALL_CALIBRATE_SAMPLES, sizeof(uint64_t), __compare_uint64);
+    /* Use 30% percentile (median) for "average". */
```

Comment:
Why this number is picked? Can we just use the median?

Reply:
Of course, this is not a 100% calculated result but it comes from experiments that I did. There are a few reasons for this:
1. The distribution of values is such that outliers tend to concentrate on the higher end of the range. This is because context switches generate big numbers. But we can't possibly get "too small" numbers. Numbers at the lower end are all good by nature. But the might be harder to get later on.
2. I could use even smaller percentile to get better precision but in experiments I saw that this range is good enough to get a reasonably precise result relatively quickly.

Ideally, we might use the smallest number that we have in the array and get the most precise result but potentially it could be unachievable in a reasonable number of attempts. Hence this trade-off between precision and speed.